src/main/utility/hash.rs

```diff
@@ -0,0 +1,18 @@
+mod export {
+    use std::collections::hash_map::DefaultHasher;
+    use std::ffi::CStr;
+    use std::hash::Hasher;
+
+    /// An easy way for C code to hash a string.
```

Comment:
I think we could also just use `g_str_hash` from glib in `c` code for now if we wanted to avoid exporting another function.

Reply:
Good idea, I didn't know about that function.